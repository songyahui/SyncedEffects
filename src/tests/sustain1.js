"use hopscript"

const hh = require("hiphop");

const prg =
      <hh.module I=${{accessibility: hh.IN}} J K>
	<hh.loop>
	  <hh.abort I>
	    <hh.sustain J/>
	  </hh.abort>
	  <hh.emit K/>
	</hh.loop>
      </hh.module>;

exports.prg = new hh.ReactiveMachine(prg, "sustain1");
