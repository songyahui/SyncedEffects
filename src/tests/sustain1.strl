module sustain1:

input I;
output J, K;

loop
	abort
		sustain J;
	when I;
	emit K;
end loop

end module