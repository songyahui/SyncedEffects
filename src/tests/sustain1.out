sustain1> ;
--- Output: J
sustain1> ;
--- Output: J
sustain1> ;
--- Output: J
sustain1> I;
--- Output: J K
sustain1> ;
--- Output: J
sustain1> ;
--- Output: J
sustain1> I;
--- Output: J K
sustain1> I;
--- Output: J K
sustain1> I;
--- Output: J K
sustain1> ;
--- Output: J
sustain1> ;
--- Output: J
